docs/illink-options.md

```diff
@@ -0,0 +1,184 @@
+
+### Linking from the main assembly
+
+The command:
+
+`illink -a Program.exe`
+
+will use the assembly Program.exe as a source. That means that the linker will
+walk through all the methods of Program.exe to generate only what is necessary
+for this assembly to run.
+
+### Linking from an [XML descriptor](data-formats.md#descriptor-format)
+
+The command:
+
+`illink -x desc.xml`
+
+will use the XML descriptor as a source. That means that the linker will
+use this file to decide what to link in a set of assemblies. The format of the
+descriptors is described further on in this document.
+
+### Actions on the assemblies
+
+You can specify what the linker should do exactly per assembly.
+
+The linker can do the following things on all or individual assemblies
+
+- `skip` - skip them, and do nothing with them
+- `copy` - copy them to the output directory
+- `copyused` - copy used assemblies to the output directory
+- `link` - link them to reduce their size
+- `delete`- remove them from the output
+- `save` - save them in memory without linking
+- `AddBypassNGen` - add BypassNGenAttribute to unmarked methods
+- `AddBypassNgenUsed` - add BypassNGenAttribute to unmarked methods in used assemblies
+
+You can specify an action per assembly using `-p` option like this:
+
+`illink -p link Foo`
+
+or
+
+`illink -p skip System.Windows.Forms`
+
+Or you can specify what to do for the core assemblies.
+
+Core assemblies are the assemblies that belong to the base class library,
+like mscorlib.dll, System.dll or System.Windows.Forms.dll.
+
+You can specify what action to do on the core assemblies with the option:
+
+`-c skip|copy|link`
+
+### The output directory
+
+By default, the linker will create an `output` directory in the current
+directory where it will store the processed files, to avoid overwritting input
+assemblies. You can change the output directory with the option:
+
+`-out PATH`
+
+If you specify the ouput directory `.', please ensure that you won't write over
+important assemblies of yours.
+
+### Specifying assemblies look up paths
+
+By default, the linker will first look for assemblies in the directories `.`
+and `bin`. You can specify additional locations where assemblies will be searched
+for by using `-d PATH` option.
+
+Example:
+
+`illink -d ../../libs -a program.exe`
+
+### Excluding framework features
+
+One of the ways how to reduce core assemblies size is by removing framework capabilities. This
+is usually something that the developer decides about as it alters the program behaviour
+therefore the decision whether the size saving is worth it is left to the developer.
+
+Each feature can be controlled independly using `--feature NAME value` option.
+
+The list of available features names is the framework dependant and can be different for different
+versions. 
+
+The list of controllable features for .NET Core is available at https://docs.microsoft.com/en-us/dotnet/core/run-time-config/.
```

Comment:
This isn't true currently, and it won't be unless we have one link-time setting per runtimeconfig setting. I'd suggest leaving this out and we can establish a new doc page describing those with linker support.